ctx.go

```diff
@@ -365,6 +365,41 @@ func (c *netHttpContext[B, P]) Body() (B, error) {
 	return body, err
 }
 
+// setParamValue sets a value to a reflect.Value based on its kind
+func setParamValue(value reflect.Value, paramValue string, kind reflect.Kind) error {
+	switch kind {
+	case reflect.String:
+		value.SetString(paramValue)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intValue, err := strconv.Atoi(paramValue)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to int: %w", paramValue, err)
+		}
+		value.SetInt(int64(intValue))
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		uintValue, err := strconv.ParseUint(paramValue, 10, 64)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to uint: %w", paramValue, err)
+		}
+		value.SetUint(uintValue)
```

Comment:
Here you are facing a problem with overflow

You should do this.

```suggestion
	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
		uintValue, err := strconv.ParseUint(paramValue, 10, bitSize(kind))
		if err != nil {
			return fmt.Errorf("cannot convert %s to uint: %w", paramValue, err)
		}
		value.SetUint(uintValue)
```

With a method that returns you the bit size

```go
func bitSize(kind) {
switch kind {
   case reflect.Uint8, reflect.Int8:
     return 8
   case reflect.Uint16, reflect.Int16:
     return 16
   case reflect.Uint32, reflect.Int32, reflect.Float32:
     return 32
   case reflect.Uint, reflect.Int:
     return strconv.IntSize()
}
return 64
}
```

I wrote my own lib about this issue

https://github.com/ccoVeille/go-safecast

I don't expect you to use it, it's just the fact it's not easy to deal with this.